packages/broker/test/integration/plugins/operator/MaintainOperatorValueService.test.ts

```diff
@@ -1,125 +1,96 @@
+        await (await token.connect(operatorWallet).transferAndCall(operatorContract.address, parseEther('200'), operatorWallet.address)).wait()
+        await (await operatorContract.stake(sponsorship1.address, parseEther('100'))).wait()
 
-        const approxValuesBefore = (await operatorContract.getApproximatePoolValuesPerSponsorship()).approxValues
-        for (const approxValue of approxValuesBefore) {
-            logger.debug(`approxValue: ${approxValue.toString()}`)
-        }
+        const sponsorship2 = await deploySponsorship(chainConfig, operatorWallet, { streamId, earningsPerSecond: parseEther('2') })
+        await (await token.connect(operatorWallet).transferAndCall(sponsorship2.address, parseEther('250'), '0x')).wait()
+        await (await operatorContract.stake(sponsorship2.address, parseEther('100'))).wait()
 
-        await waitForCondition(async () => {
-            const diff = await getDiffBetweenApproxAndRealValues()
-            const poolValue = await operatorContract.totalValueInSponsorshipsWei()
-            const threshold = penaltyFraction.mul(poolValue).div(parseEther('1')).toBigInt()
-            logger.debug(`diff: ${diff}, threshold: ${threshold}`)
-            return diff > threshold 
-        }, 10000, 1000)
-
-        await maintainOperatorValueService.start()
-
-        await waitForCondition(async () => (await operatorContract.totalValueInSponsorshipsWei()).gt(totalValueInSponsorshipsBefore), 10000, 1000)
-        
-        const diff = await getDiffBetweenApproxAndRealValues()
-
-        const poolValue = await operatorContract.totalValueInSponsorshipsWei()
-        const threshold = penaltyFraction.mul(poolValue).div(parseEther('1')).toBigInt()
-
-        expect((await operatorContract.totalValueInSponsorshipsWei()).toBigInt()).toBeGreaterThan(totalValueInSponsorshipsBefore.toBigInt())
-        logger.debug(`at end diff: ${diff}, threshold: ${threshold}`)
-        expect(diff).toBeLessThan(threshold)
-        const approxValuesAfter = (await operatorContract.getApproximatePoolValuesPerSponsorship()).approxValues
-        for (const approxValue of approxValuesAfter) {
-            logger.debug(`approxValue: ${approxValue.toString()}`)
-        }
-        // one of the values should have increased, but not both
-        expect((approxValuesAfter[0].toBigInt() > approxValuesBefore[0].toBigInt()
-            || approxValuesAfter[1].toBigInt() > approxValuesBefore[1].toBigInt())
-            && !((approxValuesAfter[0].toBigInt() > approxValuesBefore[0].toBigInt()
-            && approxValuesAfter[1].toBigInt() > approxValuesBefore[1].toBigInt()))).toBeTruthy()
+        // 1000 = check every second
+        const service = new MaintainOperatorValueService(operatorConfig, 0.5, 1000)
+
+        const poolValueBeforeWithdraw = await operatorContract.getApproximatePoolValue()
+
+        // safe threshold is 5 DATA, that's when withdraw happens
```

Comment:
Which calculation or config defines that value of `5`? (Same for value `3` at lines `87`-`88`)

Reply:
TODO: add more comments.

5 DATA = 50% of 10 DATA, 50% is the "safe threshold" (we don't want to wait all the way to the limit, lest we be late)
10 DATA = 10% of 100 DATA, 10% is the penalty limit (sum of unwithdrawn earnings may not exceed it) 
100 DATA = pool value

Reply:
Added more comments